src/QAT/qat/purr/compiler/builders.py

```diff
@@ -2,7 +2,12 @@
 # Copyright (c) 2023 Oxford Quantum Circuits Ltd
 import itertools
 import math
-from collections import defaultdict
+import sys
+
+if sys.version_info < (3, 10):
```

Comment:
why do we need this?

```
kristianb@jamie-xps-15-9520:~/Documents/code/scrapyard/pyt39_tes$ poetry run python
Python 3.8.10 (default, Nov 22 2023, 10:22:35) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections.abc import Iterable
>>> 
```

Reply:
Interesting. It fails import that way on my machine and in CI. :thinking:

Reply:
As seen here.
https://github.com/oqc-community/qat/actions/runs/9743002674/job/26885510222#step:5:334